Modules/_json.c

```diff
@@ -1660,17 +1672,25 @@ encoder_listencode_dict(PyEncoderObject *s, PyUnicodeWriter *writer,
     }
 
     if (s->markers != Py_None) {
-        int has_key;
+        Py_ssize_t len;
+
         ident = PyLong_FromVoidPtr(dct);
         if (ident == NULL)
+            return -1;
```

Comment:
```suggestion
            goto bail;
```

This doesn't leak anything but if we were to change this function, we need to update this path as well. Better to just jump to the end.